astroquery/eso/core.py

```diff
@@ -557,80 +601,164 @@ def get_headers(self, product_ids, *, cache=True):
+        with open(filename, "wb") as fd:
+            fd.write(payload)
+
+    def find_associated_files(self, datasets: List[str], *, mode: str = "raw",
+                              savexml: bool = False, destination: str = None) -> List[str]:
+        """
+        Invoke Calselector service to find calibration files associated to the provided datasets.
+
+        Parameters
+        ----------
+        datasets :
+            List of datasets for which calibration files should be retrieved.
+        mode :
+            Calselector mode: 'raw' (default) for raw calibrations,
+             or 'processed' for processed calibrations.
+        savexml :
+            If true, save to disk the XML association tree returned by Calselector.
+        destination :
+            Directory where the XML files are saved (default = astropy cache).
+
+        Returns
+        -------
+        files :
+            List of unique datasets associated to the input datasets.
+        """
+        mode = "Raw2Raw" if mode == "raw" else "Raw2Master"
+        post_data = {"dp_id": datasets, "mode": mode}
+        response = self._session.post(self.CALSELECTOR_URL, data=post_data)
+        response.raise_for_status()
+        associated_files = set()
+        content_type = response.headers['Content-Type']
+        # Calselector can return one or more XML association trees depending on the input.
+        # For a single dataset it returns one XML and content-type = 'application/xml'
+        if 'application/xml' in content_type:
+            filename = self._get_filename_from_response(response)
+            xml = response.content
+            associated_files.update(self._get_unique_files_from_association_tree(xml))
+            if savexml:
+                self._save_xml(xml, filename, destination)
+        # For multiple datasets it returns a multipart message
+        elif 'multipart/form-data' in content_type:
+            msg = email.message_from_string(f'Content-Type: {content_type}\r\n' + response.content.decode())
+            for part in msg.get_payload():
+                filename = part.get_filename()
+                xml = part.get_payload(decode=True)
+                associated_files.update(self._get_unique_files_from_association_tree(xml))
+                if savexml:
+                    self._save_xml(xml, filename, destination)
+        else:
+            raise CalSelectorError(f"Unexpected content-type '{content_type}' for {response.url}")
 
-        return resp
+        # remove input datasets from calselector results
+        return list(associated_files.difference(set(datasets)))
 
-    def retrieve_data(self, datasets, *, continuation=False, destination=None,
-                      with_calib='none', request_all_objects=False,
-                      unzip=True, request_id=None):
+    def retrieve_data(self, datasets, *, overwrite=False, destination=None,
+                      with_calib=None, unzip=True):
```

Comment:
I think this is the only remaining question that I would like to address before merging. Given ``request_id`` was featured in an example in the documentation, I'm somewhat inclined towards adding a deprecation warning for all the removed parameters. 
But I'm also OK if you say it has to be a clean cut for the users, I would really just like to see it's a decision whether giving warnings or exception and not a side effect of the code refactoring. 
(knowing that in practice warnings are mostly ignored by users, with the deprecation warning they could run something that has no effect any more, while with the exception they face the changes straight away).

Reply:
I've restored the original API and added a deprecation warning using `deprecated_renamed_argument` decorator from astropy, as done for example in `irsa/core.py`.
BTW I think the `since` argument should be corrected in `irsa/core.py` at line 63 and replaced with an array of 3 elements since there are 3 changed arguments.